source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/typeaheadsearch/TypeAheadSearchActivity.kt

```diff
@@ -0,0 +1,225 @@
+            backIconParams = typeAheadParams.getBackIconParams()
+            screenTitle = typeAheadParams.getScreenTitle()
+
+            startSearchingIconParams = typeAheadParams.getStartSearchingStateParams()
+            errorIconParams = typeAheadParams.getErrorStateParams()
+            noResultIconParams = typeAheadParams.getNoResultStateParams()
+        }
+        else {
+            crossIconParams = CrossIconParams()
+            searchIconParams = SearchIconParams()
+            tickIconParams = TickIconParams()
+            backIconParams = BackIconParams()
+            screenTitle = "Search"
+
+            startSearchingIconParams = StartSearchingStateParams()
+            errorIconParams = ErrorStateParams()
+            noResultIconParams = NoResultStateParams()
+        }
+
+        // pass choices data, host communication interface
+        viewModel.init(titleList, valueList, dataType, dataset)
+
+        // set theme
+        layoutInflater.context.setTheme(R.style.adaptiveCardStyling)
+
+        activityTypeAheadSearchBinding = ActivityTypeAheadSearchBinding.inflate(layoutInflater)
+        activityTypeAheadSearchBinding?.let { activityTypeAheadSearchBinding ->
+            searchTextView = activityTypeAheadSearchBinding.typeAheadSearchQuery
+            recyclerView = activityTypeAheadSearchBinding.searchResultsList
+            progressBarView = activityTypeAheadSearchBinding.customOverlayView
+            clearTextIconView = activityTypeAheadSearchBinding.clearTextIcon
+            descriptiveImageView = activityTypeAheadSearchBinding.errorImage
+            descriptiveTextView = activityTypeAheadSearchBinding.errorMsgText
+            activityTypeAheadSearchBinding.viewModel = viewModel
+        }
+
+        val processClearTextIconVisibility = Observer<String?> { queryText ->
+            clearTextIconView.visibility = if (queryText.isNullOrEmpty()) View.GONE else View.VISIBLE
+        }
+
+        viewModel.uiState.observe(this, ::processState)
+        viewModel.queryText.observe(this, processClearTextIconVisibility)
+
+        clearTextIconView.setBackgroundResource(crossIconParams.drawableResourceId)
+        clearTextIconView.setOnClickListener { viewModel.clearText().also { searchTextView.text.clear() } }
+        activityTypeAheadSearchBinding?.searchIcon?.setBackgroundResource(searchIconParams.drawableResourceId)
+
+        with(searchTextView) {
+            // request focus to the SearchBarView for accessibility once the activity is open.
+            isFocusable = true
+            sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED)
+            addTextChangedListener(
+                DebouncingTextWatcher(
+                    lifecycleScope,
+                    { s: String ->
+                        viewModel.fetchDynamicOptions(s)
+                    },
+                    250 // TODO: pass value from host config here
```

Comment:
declare this as a const to make it easier to spot in future iterations

Reply:
done